Sources/DatadogSDKTesting/Utils/SwiftExtensions.swift

```diff
@@ -226,7 +251,39 @@ extension Dictionary where Key == String, Value == String {
 }
 
 extension Bundle {
+    @inlinable
     var name: String {
         bundleURL.deletingPathExtension().lastPathComponent
     }
 }
+
+extension Sequence where Iterator.Element: Hashable {
+    @inlinable
+    var asSet: Set<Element> { Set(self) }
+    
+    func unique() -> [Iterator.Element] {
+        var seen: Set<Iterator.Element> = []
+        return filter { seen.insert($0).inserted }
+    }
+}
+
+extension Dictionary where Value: AnyObject {
+    mutating func get(key: Key, or create: @autoclosure () throws -> Value) rethrows -> Value {
+        if let value = self[key] { return value }
+        let value = try create()
+        self[key] = value
+        return value
+    }
+}
+
+extension Dictionary {
+    mutating func get<R>(key: Key,
+                         or create: @autoclosure () throws -> Value,
+                         _ cb: (inout Value) throws -> R) rethrows -> R
```

Comment:
formatting nit: not sure which formatting is more commonly followed in this repo, but I'd suggest breaking the line before the `key: Key` and after `throws -> R`